### DISCUSSION TITLE
Saving some custom options to a local `.npmrc` file automatically

### DISCUSSION THREAD

**Discussion by zkochan** [pnpm/pnpm#3171] [Ideas]
There are some npm/pnpm configs that change the way the `node_modules` is built.

- ignore-scripts
- no-optional
- no-bin-links
- global-style (not used by pnpm)
- legacy-bundling (not used by pnpm)

and some pnpm-specific ones might get introduced. Like ~copying instead of hard linking~ and changing the way leaf dependencies are linked to node_modules.

I suggest in order to have consistent installs for every developer, to automatically add or update an `.npmrc` file in the root of the package, with values of such configs (if their value is not the default)

**Comment by wmertens** [comment]
I wonder though if it is ok for pnpm-specific options to live in `.npmrc` (it's not called `.pnpmprc` after all). Other projects simply use `package.json`.

**Comment by zkochan** [comment]
It is a complex problem. I would also like a separate config but that would
mean additional fs operations. We should read npm configs in any case
because pnpm tends to be a drop-in replacement that works with existing
configs.

**Comment by wmertens** [comment]
well, generally, pnpm is already writing to package.json when installing
deps…

as a start, the key "pnpm" could be read at least?
